graphql-wai/src/GraphQL/Wai.hs

```diff
@@ -0,0 +1,35 @@
+{-# LANGUAGE AllowAmbiguousTypes #-}
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module GraphQL.Wai
+  ( toApplication
+  ) where
+
+import Protolude
+
+import GraphQL (interpretAnonymousQuery)
+import GraphQL.Resolver (HasResolver, Handler)
+import Network.Wai (Application, queryString, responseLBS)
+import GraphQL.Value.ToValue (toValue)
+import Network.HTTP.Types.Header (hContentType)
+import Network.HTTP.Types.Status (status200, status400)
+import qualified Data.Aeson as Aeson
+
+
+-- | Adapt a GraphQL handler to a WAI application. This is really just
+-- to illustrate the mechanism, and not production ready at this point
+-- in time.
+--
+-- If you have a 'Cat' type and a corresponding 'catHandler' then you
+-- can use "toApplication @Cat catHandler".
+toApplication :: forall r. (HasResolver IO r) => Handler IO r -> Application
+toApplication handler = app
+  where
+    app req respond =
+      case queryString req of
+        [("query", Just query)] -> do
```

Comment:
I guess it feels a bit off for me for our opinion to override the one recommended in the official GraphQL documentation, which is surely based on experience. 

However, no intrinsic objection to providing two separate things. Happier if they can be combined into one handler, but that's not necessary. As you've pointed out elsewhere, there's not much to this, let's get it merged and then iterate.